src/librustc_trans/mir/statement.rs

```diff
@@ -27,7 +27,7 @@ impl<'a, 'tcx> MirContext<'a, 'tcx> {
         debug!("trans_statement(statement={:?})", statement);
 
         let (scope, span) = self.debug_loc(statement.source_info);
-        debuginfo::set_source_location(self, &bcx, scope, span);
+        debuginfo::set_source_location(&self.debug_context, &bcx, scope, span);
```

Comment:
We use `debug_loc` in `mir/block.rs` and use the returned span for other purposes (namely, passing to `trans_intrinsic_call`) than setting debug location, so this won't quite work.